### ISSUE TITLE
Add an endpoint to test email configuration

### ISSUE THREAD

**Issue by turt2live** [matrix-org/synapse#5486]
So I don't have to create a test account and reset its password to make sure I got the right port+domain+username+password combination.

**Comment by almereyda** [comment]
It would also be helpful to document how to investigate email sending errors. Riot responding with

> An error was encountered when sending the email (Status 500)

and

```yaml
loggers:
  synapse.storage.SQL:
    level: debug
```

can be hard to parse by human eyes, due to the high amount of asynchronous jobs.
